outputs.tf

```diff
@@ -42,6 +42,12 @@ output "bucket_storage_class" {
   value       = local.bucket_storage_class
 }
 
+output "bucket_resource_key" {
+  description = "Bucket resource key"
+  value       = ibm_resource_key.resource_key
```

Comment:
Does this work? Isnt `ibm_resource_key` using a count?

Reply:
This does work, if create_resource_key is set to false

```hcl
    "bucket_resource_key": {
      "value": [],
      "type": [
        "tuple",
        []
      ],
      "sensitive": true
```

However a deeper look at the count shows that a resource key is only created when a COS instance is created, so it only contains a key when using the root module. When using the buckets sub module it is not created and thus this PR needs a rework.

It seems reasonable to generate a resource key for accessing the instance/bucket with the bucket information.
- It already includes the instance id and guid
- It already includes s3 endpoints (based on region/location)
The options are to exclude the resource key from the buckets sub module, or create them with the bucket information so that they can be found easily with just the bucket name.